### PR TITLE
API tests for Publication API

### PR DESCRIPTION
**Pull request by Eiriknil**

### CODE REVIEW

File: tests/customers/customers.feature [hunk 1]
```diff
@@ -73,7 +73,7 @@ Feature: Tests for NVA customers API
     Then status 200
     And match response contains getCustomerIdSuccessResponse
 
-    Scenario: Get by non-existing Customer id returns status Not Found
+    Scenario: GET by non-existing Customer id returns status Not Found
```
Comment (axthosarouris) [inline tests/customers/customers.feature:76]: General comment: the identifier that we are using now is a SortableIdentifer (found in commons). It accepts UUID strings as well, so it is not a problem now, but when you will create a new resource you will not get a UUID as an identifier.

File: tests/registrations/registrations.feature [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+              registrationId = registration['identifier'];
+            }
+          });
+          return registrationId
+        }
+      """
+
+      Given url 'https://api.dev.nva.aws.unit.no/publication'
+
+    Scenario: POST returns status Created and Resource detials
```
Comment (axthosarouris) [inline tests/registrations/registrations.feature]: Requested change: It should also return a `Location` Header with the URI of the resource

File: tests/registrations/registrations.feature [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+    Scenario: GET returns status Not Found when requesting non-existing Registration
+      Given path '/' + nonExistingResourceId
+      When method GET
+      Then status 404
+      And match response.title == 'Not Found'
+      And match response.status == 404
+      And match response.detail == 'Publication not found: ' + nonExistingResourceId
+
+    Scenario: GET returns status Bad Request when requesting with invalid Uuid
```
Comment (axthosarouris) [inline tests/registrations/registrations.feature]: Requested change: ... with invalid identifier.  We do not use UUIDs anymore. We accept them but we do not use them. When we will import stuff from Cristin it is possible tha thte identifier will be only a number for the Cristin publications.

File: tests/registrations/registrations.feature [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+      Given path '/' + invalidUuid
+      When method GET
+      Then status 400
+      And match response.title == 'Bad Request'
+      And match response.status == 400
+      And match response.detail == 'Identifier is not a valid UUID: ' + invalidUuid
```
Comment (axthosarouris) [inline tests/registrations/registrations.feature]: Create a bug for it:
https://unit.atlassian.net/browse/NP-2388?atlOrigin=eyJpIjoiNTQ0MDIzZDJhNTVjNDA4ZjllNWFhZWYwNjg0Y2YyZjgiLCJwIjoiaiJ9

**Review by axthosarouris — CHANGES_REQUESTED** [review]
Requested changed have been annotated. The rest are general comments.

File: tests/registrations/registrations.feature [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+      """
+
+      Given url 'https://api.dev.nva.aws.unit.no/publication'
+
+    Scenario: POST returns status Created and Resource detials
+      Given path '/'
+      And request correctResourcePayload
+      When method POST
+      Then status 201
+
```
Comment (brinxmat) [inline tests/registrations/registrations.feature:68]: We need to cover the Problem responses that we expect, Bad GW, etc. In addition, we need to be stringent about following the IETF specs for e.g. PUT, which we do not at the moment.

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: tests/customers/customers.feature (hunk 1 above)
Comment (Eiriknil) [reply tests/customers/customers.feature:76]: Ok, tried to create a SortableIdentifier for a non-existing publication and get 
`{
  "title" : "Bad Request",
  "status" : 400,
  "detail" : "Identifier is not a valid UUID: 178450b5f4b-88859b77-2b77-4e35-a81a-2a54e373a84e",
  "requestId" : "fbcb3327-f45e-4d9b-ad5a-72271d4134b7"
}`

File: tests/registrations/registrations.feature (hunk 5 above)
Comment (Eiriknil) [reply tests/registrations/registrations.feature:68]: Agree, but I need descriptions of the expected responses if I am to make tests for this.

File: tests/customers/customers.feature (hunk 1 above)
Comment (axthosarouris) [reply tests/customers/customers.feature:76]: my bad. I thought we were talking about publication identifiers.

**Review by axthosarouris — APPROVED** [review]

File: tests/registrations/registrations.feature (hunk 5 above)
Comment (brinxmat) [reply tests/registrations/registrations.feature:68]: I'll create a task

**Review by brinxmat — APPROVED** [review]

**Pull request merged**